Source/relay/TourGuide/Sets/Sneaks.ash

```diff
@@ -0,0 +1,363 @@
+        
+        //   NAME OF SNEAKS
+        //   1 sneaksource
+        //   1 sneaksource
+
+        // Where the name is bold and the sneaksources are tiny. In some future world,
+        // it might be nice to have coloring that grays those that are not available 
+        // yet, but that is too much for this first implementation.
+        return HTMLGenerateSpanOfClass(title, "r_bold") + "<br>" + HTMLGenerateSpanOfStyle(finalDesc, "font-size:0.8em");
+    }
+
+    // We aren't going to do a loop here; we're just going to populate table_lines
+    //   semi-manually, using logic that is roughly correct in each case.
+
+    // START = DELAY SNEAKS
+    //   1x hidden apartment
+    //   1x hidden office
+
+    // Initialize your string-int array of the sneaks.
+    string [int] sneakDelay;
+
+    // You want to use prefs when possible to isolate that the user can use that sneak, then append to sneakDelay
+    if (get_property_int("hiddenApartmentProgress") < 7) sneakDelay.listAppend("1 hidden apartment");
+    if (get_property_int("hiddenOfficeProgress") < 7) sneakDelay.listAppend("1 hidden office");
+
+    // Populate the sneaky table.
+    tableLines[1] = populateSneakTable("Delay Zones", sneakDelay);
+
+    // Add the detected NCs to your total NCs remaining.
+    totalNCsRemaining += sneakDelay.count();
+
+    // NEXT = 95% COMBAT SNEAKS
+    //   12x friars NCs (-1 w/ carto)
+    //    1x castle basement
+    //    1x castle top
+
+    string [int] sneak95;
+
+    // Ripping some code from the friars tile to count NCs encountered. First, names of the relevant NCs.
+    boolean [string] necks_known_ncs = $strings[How Do We Do It? Quaint and Curious Volume!,Strike One!,Olive My Love To You\, Oh.,Dodecahedrariffic!];
+    boolean [string] heart_known_ncs = $strings[Moon Over the Dark Heart,Running the Lode,I\, Martin,Imp Be Nimble\, Imp Be Quick];
+    boolean [string] elbow_known_ncs = $strings[Deep Imp Act,Imp Art\, Some Wisdom,A Secret\, But Not the Secret You're Looking For,Butter Knife? I'll Take the Knife];
+    
+    // Then, a tiny function to count the NCs found by zone for friars.
+    int countFriarNCs(boolean [string] known_ncs, location place) {
+        int ncs_found = 0;
+
+        if (known_ncs.count() > 0) {
+            string [int] location_ncs = place.locationSeenNoncombats();
+
+            foreach key, s in location_ncs
+            {
+                if (known_ncs contains s) ncs_found += 1;
+            }
+        }
+
+        return ncs_found;
+
+    }
+
+    // You can remove one from the needed NCs if they have carto in their run.
+    int cartoAdjustment = lookupSkill("Comprehensive Cartography").have_skill() ? 1 : 0;
+
+    // Right now I think this is off; I noted in the discord that we are having some small issues with
+    //   it showing extra NCs in a few places. Not really sure what's up with that? An easy fix is to
+    //   just set all of these to 0 in the event that questL06Friar = 'finished'. 
```

Comment:
IMO we don't need to improve this before the PR gets merged, but it would feel good to figure out the right way to handle all this logic better in the actual friar's quest so we could reuse it everywhere instead of having logic spread through various other tiles.

Reply:
agreed. i think one of the core issues is that the current pref structure for this quest is just super weird.

still -- my solution is that if you have finished the quests, the new min statement will minimize your NCs remaining to 0. i do not love this solution but it works in a pinch.